lib/OpenQA/Utils.pm

```diff
@@ -164,6 +166,35 @@ sub testcasedir {
     return $dir;
 }
 
+sub gitrepodir {
+    my %args = (
+        distri  => '',
+        version => '',
+        repo    => 'tests',
+        @_,
+    );
+    my $path
+      = $args{repo} eq 'needles'
+      ? needledir($args{distri}, $args{version})
+      : testcasedir($args{distri}, $args{version});
+    my $filename = '';
+    if (-e path($path, '.git')) {
+        $filename = path($path, '.git', 'config');
+    }
+    else { log_warning("$path is not a git directory"); }
```

Comment:
does it make sense to actually call this function if $path is not a git directory?

Reply:
you mean the `log_warning`? wouldnt be useful info? maybe for debugging? actually i feel guilty not adding more logs.

Reply:
No I mean does it make sense to call the complete function "gitrepodir" when the path "$path/.git" does not even exist when you would end up in this branch

Reply:
i see. the function does the check itself. i dont see the problem with that. i tried to keep it simple and practical. this is called with `_show` and the results are assigned to `testgiturl` && `needlegiturl`. from here you can do what ever you want. has no side effect.